inst/tests/tests.Rraw

```diff
@@ -21084,3 +21084,23 @@ test(2307, { capture.output(print(DT, class = TRUE, show.indices = TRUE)); TRUE
 dt = data.table(date=as.IDate(c(NA, "2014-12-05")))
 test(2308.01, fread("date\nNA\n2014-12-05", keepLeadingZeros=TRUE), dt)
 test(2308.02, fread("date\nNA\n2014-12-05", keepLeadingZeros=FALSE), dt)
+
+# Test that as.data.table.data.table preserves key when explicitly specified but not when omitted
+# data.frame to data.table with key
+test(2309.01, {
+  key(as.data.table(data.frame(t = c(3:1, 4:5), y = 1:5), key="t"))
+}, "t")
+# tibble to data.table with key
+DF <- data.frame(t = c(3:1, 4:5), y = 1:5)
```

Comment:
```suggestion
DF = data.frame(t = c(3:1, 4:5), y = 1:5)
test(2309.01, key(as.data.table(DF, key="t")), "t")
```

Please keep to our style of tests, e.g. setting it up and then test on a single line. Also keep in mind that we do not use `<-` when not absolutely necessary.